src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxQueryBuilder.cs

```diff
@@ -614,844 +625,2111 @@ IQueryMatch HandleStringBetween()
-                            new SortingMatch.AlphanumericDescendingMatchComparer(indexSearcher, firstFieldId, firstOrderTypeField),
-                            new SortingMatch.AlphanumericDescendingMatchComparer(indexSearcher, secondFieldId, secondOrderTypeField),
-                            new SortingMatch.AlphanumericAscendingMatchComparer(indexSearcher, thirdFieldId, thirdOrderTypeField)),
-                        (QueryBuilderHelper.ComparerType.DescendingAlphanumeric, QueryBuilderHelper.ComparerType.DescendingAlphanumeric, QueryBuilderHelper.ComparerType
-                            .DescendingAlphanumeric) => SortingMultiMatch.Create(
-                            indexSearcher, match,
-                            new SortingMatch.AlphanumericDescendingMatchComparer(indexSearcher, firstFieldId, firstOrderTypeField),
-                            new SortingMatch.AlphanumericDescendingMatchComparer(indexSearcher, secondFieldId, secondOrderTypeField),
-                            new SortingMatch.AlphanumericDescendingMatchComparer(indexSearcher, thirdFieldId, thirdOrderTypeField)),
+                        (QueryBuilderHelper.ComparerType.DescendingAlphanumeric, QueryBuilderHelper.ComparerType.DescendingSpatial, QueryBuilderHelper.ComparerType
+                            .AscendingSpatial) => SortingMultiMatch.Create(indexSearcher, match,
+                            new SortingMatch.AlphanumericDescendingMatchComparer(indexSearcher, orderMetadata[0].FieldId, orderMetadata[0].FieldType),
+                            new SortingMatch.SpatialDescendingMatchComparer(indexSearcher, in orderMetadata[1]),
+                            new SortingMatch.SpatialAscendingMatchComparer(indexSearcher, in orderMetadata[2])),
+                        (QueryBuilderHelper.ComparerType.DescendingAlphanumeric, QueryBuilderHelper.ComparerType.DescendingSpatial, QueryBuilderHelper.ComparerType
+                            .DescendingSpatial) => SortingMultiMatch.Create(indexSearcher, match,
+                            new SortingMatch.AlphanumericDescendingMatchComparer(indexSearcher, orderMetadata[0].FieldId, orderMetadata[0].FieldType),
+                            new SortingMatch.SpatialDescendingMatchComparer(indexSearcher, in orderMetadata[1]),
+                            new SortingMatch.SpatialDescendingMatchComparer(indexSearcher, in orderMetadata[2])),
+                        (QueryBuilderHelper.ComparerType.Boosting, QueryBuilderHelper.ComparerType.Ascending, QueryBuilderHelper.ComparerType.Ascending) =>
+                            SortingMultiMatch.Create(indexSearcher, match,
+                                default(BoostingComparer),
+                                new SortingMatch.AscendingMatchComparer(indexSearcher, orderMetadata[1].FieldId, orderMetadata[1].FieldType),
+                                new SortingMatch.AscendingMatchComparer(indexSearcher, orderMetadata[2].FieldId, orderMetadata[2].FieldType)),
```

Comment:
Note: this code was generated by the following snippet:

https://issues.hibernatingrhinos.com/issue/RavenDB-17376#focus=Comments-67-353226.0-0